### ISSUE TITLE
[Bug] Unwanted rerender from side effects with "createOrder" prop

### ISSUE THREAD

**Issue by yocmen** [paypal/react-paypal-js#55]
### 🐞 Describe the Bug

The popup windows just close without showing an error

### 🔬 Minimal Reproduction

On my react component I have:

```
const [internalOrderNumber, setinternalOrderNumber] = useState(null);
```

then I have a method *handleCreateOrder* to manage the create order on my server-side:

```
const handleCreateOrder = async() => {
    const body = {};
    const { error, result } = await createOrder(body);

    setinternalOrderNumber(result.internalOrderNumber); // My own order number
    return result.orderID; //paypal order number
}
```

### 😕 Actual Behavior

If I use the `setinternalOrderNumber(result.internalOrderNumber); `  the popup just banished without message

### 🤔 Expected Behavior

But if I remove the `setinternalOrderNumber(result.internalOrderNumber); `  line it works like be expected.

### 🌍 Environment

GatsbyJS

### ➕ Additional Context

I have client-side and server-side code to perform the payments, in this case, is just the client-side.

Can I manage states from the Button methods? if is possible how can I handle that? thanks in advance for all the help.

**Comment by gregjopa** [comment]
Hi @yocmen, thanks for creating this issue. I was able to reproduce this problem with client-side only code:

```js
    const [orderId, setOrderId] = useState(null);

    function createOrder(data, actions) {
        return actions.order.create({
            purchase_units: [
                {
                    amount: {
                        value: amount,
                    },
                },
            ],
        }).then(orderId => {
            setOrderId(orderId);
            return orderId;
        })
    }

// and later on render the component
<PayPalButtons createOrder={createOrder} />
```

Right now the `useEffect()` hook for `<PayPalButtons />` rerenders anytime props change. Since `createOrder` is a function, any state changes from the callback execution seem to trigger a rerender. There are use cases like having a dynamic amount where we want this behavior: https://paypal.github.io/react-paypal-js/?path=/story/example-paypalbuttons--dynamic-amount. But for this use case where `createOrder()` tracks orderId in state,  we do not want it to trigger a rerender. I'm not exactly sure how to support both use cases 🤔 I'm open to ideas/recommendations on this.

In the mean time, your best bet is to create a custom version of PayPalButtons.js and limit what you track in the [useEffect() hook](https://github.com/paypal/react-paypal-js/blob/main/src/components/PayPalButtons.js#L20-L45). For example, you could update the useEffect hook to only track `isResolved` and certain props:

```js
useEffect(() => {
// use the second param of useEffect to define what state changes should trigger a rerender
}, [isResolved, props.currency]);
```

**Comment by yocmen** [comment]
Ok, I will do some test and try to find a solution for both cases, if I found something I will let you know with a PR, really thanks for this job, Paypal need more projects like this.

**Comment by gregjopa** [comment]
This bug was fixed in #56 and is included in the v3 release.
